ring-devel/test/ring/middleware/test/stacktrace.clj

```diff
@@ -1,20 +1,20 @@
 (ns ring.middleware.test.stacktrace
-  (:require [clojure.test :refer :all]
-            [ring.middleware.stacktrace :refer :all]))
+  (:require [clojure.test :refer [deftest is testing]]
```

Comment:
All of this is good, except that I'd prefer to keep `:all` here, as that matches the style in other tests.

Reply:
Do you have a rule for when to use `:all`?  It seem that other middleware use a mix of `:all` and `[function]`.  Some of the tests use the `[ring.util.io :refer [string-input-stream]]` for project specific imports.

Reply:
For recent namespaces, I've tried to only use `:refer :all` for `clojure.test`. Namespaces that differ to this should be changed. I would also accept a PR that removes *all* use of `:refer :all`, but I'd rather that be done as one big change across all the test namespaces, in order to ensure consistency.

Reply:
After we get through the changes im hoping for in the stack trace, I dont mind putting that PR together.